### PR TITLE
Display count of deferred apps on monthly stats

### PR DESCRIPTION
**Pull request by stevehook**
## Context

The deferred applications counter in the _Footnote_ section of the monthly statistics report is hard-coded. This PR adds a helper to `MonthlyStatisticsReport` to calculate the correct number to render.

## Changes proposed in this pull request

- Added MonthlyStatisticsReport#deferred_application_count helper.
- Display this in the view template and handle different cases (0, 1 or
  more deferred applications) with appropriate content.

#### when there are no deferred applications (or data is missing):
image

#### when there is one deferred application:
image

#### when there are multiple deferred application:
image

## Guidance to review

- Is the query that fetches the number correct? (I found only 1 application in the current cycle that matched).
- Should this number be 'cached' as part of the `MonthlyStatisticsReport` record so that it represents the correct number at the time the report was generated?
- What changes do we need to make to the copy to (a) make it clearer (b) work next year when UCAS wasn't involved in the previous cycle?

## Link to Trello card

https://trello.com/c/Eo5jeO7a/4214-work-out-what-to-do-with-monthly-stats-footnotes

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by EmmaFrith** [comment]
I've only got as far as looking at the content in the first screenshot so far - do you think this makes it clearer:

---

Deferred candidates are counted in a separate column in the tables.

They’re counted in the statistics report for the academic year in which they start their training, rather than the year in which they apply. 

This means that anyone who applied this year but deferring until next year will be counted in next year’s report, rather than this one. 

Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training. 

---

Will take a look at the other bits later today

**Comment by EmmaFrith** [comment]
Is the second screenshot trying to say the following? @stevehook 

---

Deferred candidates are counted in a separate column in the tables.

There is one deferred candidate in this data report that was *also* counted in last year’s publication by UCAS. 

This is because UCAS counted deferred candidates in the statistics report for the academic year in which they applied, rather than the year in which they start training. 

This report, on the other hand, counts deferred candidates in the year in which they start their training, rather than the year in which they apply. 

This means that anyone who applied this year but deferring until next year will be counted in next year’s report, rather than this one. 

Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training. 

---

**Comment by EmmaFrith** [comment]
Is the last screenshot trying to say the following? @stevehook 

---

Deferred candidates are counted in a separate column in the tables.

There are 24 deferred candidates in this data report that were *also* counted in last year’s publication by UCAS. 

This is because UCAS counted deferred candidates in the statistics report for the academic year in which they applied, rather than the year in which they start training. 

This report, on the other hand, counts deferred candidates in the year in which they start their training, rather than the year in which they apply. 

Anyone who applied this year but deferring until next year will be counted in next year’s report, rather than this one. 

Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training. 

---

**Comment by stevehook** [comment]
> I've only got as far as looking at the content in the first screenshot so far - is this what it's trying to say? @stevehook :
> 
> Deferred candidates are counted in a separate column in the tables.
> 
> They’re counted in the statistics report for the academic year in which they start their training, rather than the year in which they apply.
> 
> This means that anyone who applied this year but deferring until next year will be counted in next year’s report, rather than this one.
> 
> Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training.
> 
> Will take a look at the other bits later today


Thanks @EmmaFrith, I think it's a bit more like this (apologies for mangling the words)...

Deferred candidates are counted in a separate column in the tables.
 
They’re counted in the statistics report for the academic year in which they start their training _after their offer is re-confirmed_, _as well as_ the year in which they apply.
 
This means that anyone who applied this year but deferring until next year will be counted in next year’s report, _as well as_ this one.
 
Deferred candidates are only counted _in the academic year that they start their course in_ once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training.

**Comment by EmmaFrith** [comment]
> > I've only got as far as looking at the content in the first screenshot so far - is this what it's trying to say? @stevehook :
> > Deferred candidates are counted in a separate column in the tables.
> > They’re counted in the statistics report for the academic year in which they start their training, rather than the year in which they apply.
> > This means that anyone who applied this year but deferring until next year will be counted in next year’s report, rather than this one.
> > Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training.
> > Will take a look at the other bits later today
> 
> Thanks @EmmaFrith, I think it's a bit more like this (apologies for mangling the words)...
> 
> Deferred candidates are counted in a separate column in the tables.
> 
> They’re counted in the statistics report for the academic year in which they start their training _after their offer is re-confirmed_, _as well as_ the year in which they apply.
> 
> This means that anyone who applied this year but deferring until next year will be counted in next year’s report, _as well as_ this one.
> 
> Deferred candidates are only counted _in the academic year that they start their course in_ once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training.

ah ok so they're eventually counted twice? if that's the case, how you've edited the copy here looks good to me

**Comment by MylesJarvis** [comment]
@stevehook @EmmaFrith re the question on guidance for review. Is there some value in having some if/elseif conditional logic that is driven by date. So after say, 12 October 2022 (or whenever next cycle opens), we render something that removes mention to UCAS altogether? Or do we just put a reminder to tackle that in the next cycle on the backlog and just do a PR then?

**Comment by stevehook** [comment]
@EmmaFrith I've updated the content and screenshots following with your suggestions.

**Comment by stevehook** [comment]
> @stevehook @EmmaFrith re the question on guidance for review. Is there some value in having some if/elseif conditional logic that is driven by date. So after say, 12 October 2022 (or whenever next cycle opens), we render something that removes mention to UCAS altogether? Or do we just put a reminder to tackle that in the next cycle on the backlog and just do a PR then?

Yes I think we need that. Might as well do it now, though I'm wondering whether we should be extract this into a component rather than overloading the view with conditional logic...

**Comment by EmmaFrith** [comment]
@stevehook could you change this sentence to 'Deferred candidates are only counted in the academic year *in which* they start their course once training providers re-confirm their deferred offer.'?

Screenshot 2022-01-04 at 10 22 20

**Review by EmmaFrith — APPROVED** [review]
Not sure I'm best placed to to comment on the accuracy of how we've understood the presentation of the data - but the wording in itself makes sense to me.

File: app/views/publications/monthly_statistics/show.html.erb [hunk 1]
```diff
@@ -273,6 +273,70 @@
...
+      <p class="govuk-body">
+        Deferred candidates are only counted once training providers re-confirm their deferred offer. Providers need to do this before candidates intend to start their training. 
+      </p>
+    <% elsif @presenter.deferred_applications_count > 1 %>
+      <p class="govuk-body">
+        Deferred candidates are counted in a separate column in the tables.
+      </p>
+
+      <p class="govuk-body">
+        There are <%= pluralize(@presenter.deferred_applications_count, 'deferred candidate') %> in this data report that were also counted in last year’s publication by UCAS.
```
Comment (duncanjbrown) [inline app/views/publications/monthly_statistics/show.html.erb:322]: Could we dedupe all this? We should have inflection for `is/are` (we already have `was/were`)

**Review by duncanjbrown — COMMENTED** [review]
This looks good to me. Query now returns 46 records on prod, which feels right.

File: app/views/publications/monthly_statistics/show.html.erb (hunk 1 above)
Comment (stevehook) [reply app/views/publications/monthly_statistics/show.html.erb]: Thanks @duncanjbrown - have had a go at de-duping this.

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**